xds/src/main/java/io/grpc/xds/LoadReportClient.java

```diff
@@ -340,4 +313,106 @@ private void cleanUp() {
       }
     }
   }
+
+  private final class LrsStreamV2 extends LrsStream {
+    StreamObserver<io.envoyproxy.envoy.service.load_stats.v2.LoadStatsRequest> lrsRequestWriterV2;
+
+    @Override
+    void start() {
+      StreamObserver<io.envoyproxy.envoy.service.load_stats.v2.LoadStatsResponse>
```

Comment:
I created #7341 to do that for XdsClientImpl.

Reply:
I was following [favor composition over inheritance](https://effectivejava.googleplex.com/third-edition#page=108). Even before the refactoring `LrsStream` is not used as a StreamObserver by the outer class.

Reply:
The disadvantages of inheritance usually only come with class extension but not for implementing an interface, as an interface is only a contract. It's pure, it doesn't imply any class states. This implementation is much cleaner without any other consequence, [GrpcLbState](https://github.com/grpc/grpc-java/blob/a91acec2d4429c40d3d1fe38745f712f86b0c470/grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java#L522) contains has a similar implementation and it looks great.

Now that we are doing a refactoring here, it's better to make improvement as we can see as well.

Reply:
It's not necessary to make LrsStream a StreamObserver. The outer class does not use it as StreamObserver. `LrsStream` is already a large class. It's better to have a StreamObserver object/field that is only responsible to do StreamObserver's job.

Reply:
> The outer class does not use it as StreamObserver. LrsStream is already a large class.

This argument is not strong enough not to do so. My opinion is that, if we could easily make things flat, don't make them nested. Unnecessary nesting is a bad style IMO. Also, you are able to avoid things like `LrsStreamV2.this.onError(t)`. 😄

Reply:
I don't think making things in different hierarchies flat is always a good style. `LsrStream` has a `stub` inside it and it is a `StreamObserver` in the mean time, the hierarchy is completely twisted. The response observer is a small object used locally in only one place, nesting it (as a local variable at where is it used) is better than flatting it out in a big class.

`LrsStreamV2.this.onError(t)` is just because the method name is `onError()`.

Reply:
Alright, I am not gonna block you on this. Anyway, my opinion is having `LrsStream` implement `StreamObserver` is 

1) simpler
2) You won't need `LrsStreamV2.this.onError(t)` but it just be `onError()`, although you would need to change the method name (btw the name `onError()` on the abstract class is lazy and bad =p).